File: backend/app/schemas/building.py
import datetime

from pydantic import BaseModel, ConfigDict


class BuildingBase(BaseModel):
    campus_id: int
    name: str
    code: str | None = None
    description: str | None = None
    latitude: float | None = None
    longitude: float | None = None


class BuildingCreate(BuildingBase):
    pass


class BuildingUpdate(BaseModel):
    campus_id: int | None = None
    name: str | None = None
    code: str | None = None
    description: str | None = None
    latitude: float | None = None
    longitude: float | None = None


class BuildingRead(BuildingBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: datetime.datetime
    updated_at: datetime.datetime

File: frontend/src/types/building.ts
export interface Building {
  id: number;
  campus_id: number;
  name: string;
  code: string | null;
  description: string | null;
  /** Real, surveyed building coordinates (Phase 17). Null for every
   * building until an on-site GPS survey populates them — see
   * docs/phase17_satellite_campus_map.md. Never fall back to an
   * approximated value when these are null; render nothing instead. */
  latitude: number | null;
  longitude: number | null;
  created_at: string;
  updated_at: string;
}
